docs/_posts/2019-03-05-go.md

```diff
@@ -0,0 +1,50 @@
+
+Go's high-level APIs have also been valuable to running faster development cycles. Usually in a low-level language, it can be a real pain to write hundreds of lines of code just to implement a certain function. With the high-level APIs that Go provides, most of this is concealed through abstraction.  
+
+## The toolchain 
+
+Go’s open source tools and built-in toolchain add additional functionality to projects. These tools can either be imported or implemented into any code environment as stand-alone console applications. Specifically, for the Secretless Broker, the tools we enjoy and find most helpful are `pprof` for profiling, `golint` for code clean up, and `godocs` for documentation automation. 
+
+`pprof`is a profiling tool that once added to the main function of a program, provides an in-depth analysis of processes. With Secretless, every byte of data travels through the broker before reaching and returning to its endpoints. We found that this exchange was taking too long, but we couldn’t tell exactly where the latency was coming from. With pprof, we were able visualize the effect of our project on a host machine’s CPU and memory. This is crucial because by using this tool, we gain visibility into processes and can tune them to provide a more positive experience for our customers. 
+
+It also became difficult to consistently format our code, especially as our codebase increased in size and complexity. To fix this common problem, we added the `golint` automated tool into our workflow. Code linters run source code against an applied set of stylistic and syntax rules, raising flags when this structure is broken.  Linting has been an indispensable productivity tool for writing code and has even sped up our code review process.  
+
+Documentation is essential to growing and sustaining the community around our project. We add to and tweak our README and contributor guides to encourage end-user self-efficiency and success. This has proven to be a time-consuming undertaking because our codebase is constantly evolving and being refactored. To our benefit, Go offers the `godoc` documentation tool to dynamically create documentation. This tool parses the code, relevant comments, and structure to generate documentation that evolves as fast as our code does. 
+
+## … and beyond the toolchain 
+
+With Go, we can produce statically linked binaries that fully contain our applications and run them on different platforms. This capability supported our decision to move to Docker-based development for our building and testing.  
```

Comment:
```suggestion
At an early stage of the project, we have been focusing on providing a native and secure Secretless solution for Kubernetes and Openshift environments.  To accomplish this, we needed to wrap the project inside a portable, containerized environment. Go’s ability to produce statically linked binaries that can be run across different platforms allowed us to work towards this native solution and move to Docker-based development for our building and testing.
```
```suggestion
With Go, we can produce statically linked binaries that fully contain our applications and run them on different platforms. This capability supported our decision to move to Docker-based development for our building and testing. 
```

Reply:
@typaulhus please wait for @izgeri approval on this one before merging into PR

Reply:
I don't think this is accurate:
> To accomplish this, we needed to wrap the project inside a portable, containerized environment. Go’s ability to produce statically linked binaries that can be run across different platforms allowed us to work towards this native solution and move to Docker-based development for our building and testing.

I would prefer we say:
> To accomplish this, we wanted to deliver Secretless so that it could be deployed as a sidecar container. 

and then just merge the next paragraph into this one.